### ISSUE TITLE
Undefined behavior in constructor of child of RouterElement

### ISSUE THREAD

**Issue by jbnoblot** [vigilock/fast-custom-router#10]
**Describe the bug**
```console
Argument of type 'typeof Import' is not assignable to parameter of type 'RouterElement'.
  Type 'typeof Import' is missing the following properties from type 'RouterElement': child, __parseEmptyConfig, __parseUnusedArguments, __loadModule

21     super(Import, [], {})
             ~~~~~~

  lib/models/Import.js:21:11
    21     super(Import, [], {})
                 ~~~~~~
    Did you mean to use 'new' with this expression?
```

**To Reproduce**
If I take class `Import` (same bug for all extends of `RouterElement`) in constructor call of super. 
super.constructor take a `RouterElement` as first param
But Import super call = `super(Import, [], {})` ([link direct](https://github.com/vigilock/fast-custom-router/blob/main/lib/models/Import.js#L20))

**Expected behavior**
No Idea but what the value of Import ?

**Screenshots**

**System informations (please complete the following information):**

- OS: Fedora
- Node Version 16.4.0

**Additional context**
X

**Comment by Couapy** [comment]
The [RouterElement constructor](https://github.com/vigilock/fast-custom-router/blob/main/lib/models/RouterElement.js#L15) does not expect an instance of RouterElement (or other class that inherit from this class), but only the class.

This allow to display better errors when the configuration is incorrect (only the name of the class, accessed by `Class.name`, see examples below), e.g. : 

1. [here](https://github.com/vigilock/fast-custom-router/blob/main/lib/models/RouterElement.js#L29)
2. or [here](https://github.com/vigilock/fast-custom-router/blob/main/lib/models/RouterElement.js#L43)

**Comment by jbnoblot** [comment]
Only the class ? What do you mean ?  

Yes about that only child as property name not RouterElement. but method of RouterElement use this.child.name
maybe add this property to RouterElement

**Comment by Couapy** [comment]
Yes maybe only provide the class name instead of providing class and accessing to the name should be a better idea.
